ludwig/utils/dataset_utils.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+from ludwig.constants import TEST_SPLIT, TRAIN_SPLIT, VALIDATION_SPLIT
+
+
+def get_repeatable_train_val_test_split(
+    df_input, stratify_colname, random_seed, frac_train=0.7, frac_val=0.1, frac_test=0.2
+):
+    """Return df_input with split column containing (if possible) non-zero rows in the train split, validation
+    split, and test split categories.
+
+    If the input dataframe does not contain an existing split column or if the
+    number of rows in both the validation and test split is 0, return df_input
+    with split column set according to frac_<type> and stratify_colname.
+
+    Else stratify_colname is ignored, and:
+     If the input dataframe contains an existing split column and non-zero row
+      counts for all three split types, return df_input.
+     If the input dataframe contains an existing split column but only one of
+      validation and test split has non-zero row counts, return df_input with
+      missing split getting rows from train split as per frac_<type>.
+
+    Parameters
+    ----------
+    df_input : Pandas dataframe
+        Input dataframe to be split.
+    stratify_colname : str
+        The column used for stratification; usually the label column.
+    random_seed : int
+        Seed used to get repeatable split.
+    frac_train : float
+    frac_val   : float
+    frac_test  : float
+        The ratios with which to split the dataframe into train, val, and test data;
+        should sum to 1.0.
+
+    Returns
+    -------
+    df_split :
+        Dataframe containing the three splits.
+    """
+
+    if frac_train + frac_val + frac_test != 1.0:
+        raise ValueError(f"fractions {frac_train:f}, {frac_val:f}, {frac_test:f} do not add up to 1.0")
+    if stratify_colname not in df_input.columns:
+        raise ValueError("%s is not a column in the dataframe" % (stratify_colname))
+
+    do_stratify_split = True
+    if "split" in df_input.columns:
```

Comment:
Thanks for this idea.  If the same functionality in this method can be cleanly implemented using 
global_preprocessing_parameters/split_dataset(), then it sounds like the way to go.  At this point,
I don't have enough context to understand if that is the case.  Maybe a good strategy is for me to wait
until the @tgaddair PR is finalized/landed and then I can assess how/whether that code can be used
to achieve the same results as this PR, and then respond.

Reply:
Hey @amholler, thanks for putting this together. I agree with @justinxzhao that it makes sense to use the new splitter API here. To make that easier, I added your changes to stratified splitting into the PR and added determinism tests, so hopefully it should be very similar in terms of the end result.

The PR is finalized in terms of API at this point, just cleaning up some implementation details raised in comments.

Reply:
Looking at this PR and the dataset splitting PR more closely, it looks like it would be rather complex to do an in-place replacement that uses the new dataset splitting API that would be more concise than what's already here.

This function flexibly handles cases where the split column already exists and when there may or may not be a validation or test set, which the dataset splitting API does not do. It does not necessarily work for dask dataframes, but this is also a separate goal from unblocking users referring to [Ludwig automl examples](https://ludwig-ai.github.io/ludwig-docs/0.5/user_guide/automl/).

My revised thinking is that we should check this in as is so that we can start using it in ludwig/experiments.